lib/server.js

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable prefer-template */
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-useless-escape */
+const Boom = require('@hapi/boom');
+const request = require('superagent');
+const tools = require('auth0-extension-tools');
+
+module.exports.createServer = cb => fromHapi(tools.createServer(cb));
+
+const SANITIZE_RX = /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g;
+
+function fromHapi(serverFactory) {
+  let hapiServer;
+  let webtaskContext;
+
+  return async (context, req, res) => {
+    webtaskContext = attachStorageHelpers(context);
+
+    if (hapiServer == null) {
+      hapiServer = await serverFactory(webtaskContext);
+
+      if (!hapiServer) {
+        throw new Error('Server factory did not return a server instance');
+      }
+
+      hapiServer.ext('onRequest', (hapiRequest, h) => {
+        const normalizeRouteRx = createRouteNormalizationRx(hapiRequest.raw.req.x_wt);
+
+        if (normalizeRouteRx) {
+          hapiRequest.originalUrl = hapiRequest.path;
+          // need to remove the protocol + domain etc for the route matching to work,
+          // however, need to keep the query string for it to be passed to the route handlers
+          hapiRequest.setUrl(`${hapiRequest.path}${hapiRequest.url.search}`.replace(normalizeRouteRx, '/'));
+        }
+
+        /* Fix multi-proto environments, take the first */
+        if (hapiRequest.headers['x-forwarded-proto']) {
+          hapiRequest.headers['x-forwarded-proto'] = hapiRequest.headers['x-forwarded-proto'].split(',').shift();
+        }
+
+        hapiRequest.webtaskContext = webtaskContext;
+
+        return h.continue;
+      });
+    }
+
+    hapiServer.listener.emit('request', req, res);
+  };
+}
+
+const USE_WILDCARD_DOMAIN = 3;
+const USE_CUSTOM_DOMAIN = 2;
+const USE_SHARED_DOMAIN = 1;
+
+function createRouteNormalizationRx(claims) {
```

Comment:
Seems like this is the exact same method as the one in `urlHelpers.js`, any reason we need two implementations?